source/docs/software/advanced-controls/introduction/tuning-pid-controller.rst

```diff
@@ -1,32 +1,229 @@
+   
+.. raw:: html
+
+   </details> <br>
+
+This is a case where feedback control alone is insufficient to achieve good behavior with the system.
+
+Step 2: Feed-Forward, then FeedBack
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The core reason for this is that gravity pulls on the arm in a :term:`non-linear` fashion. That is to say, the amount of :term:`torque` that gravity exerts on our arm is proportional to the *cosine* of the current angle.
+
+To counteract this, we introduce a feed-forward term which is also proportional to the cosine of the angle. 
+
+.. math::
+   V_{ff} = K_{cosFF} * cos(\theta_{arm})
+
+:math:`K_{cosFF}` could be calculated if all the mechanical and physical properties of the system are known. However, since a lot of these are hard to model accurately, we will determine it experimentally.
+
+Perform the following:
+
+1. Set :math:`K_p`, :math:`K_i`, :math:`K_d`, and :math:`K_{cosFF}` to zero.
+2. Increase and decrease :math:`K_{cosFF}` until the arm can hold its position with as little movement as possible. In this simulation, you'll want to go out to at least four decimal points.
+3. Increase :math:`K_p` until the :term:`output` starts approaches the :term:`setpoint`.
+4. Increase :math:`K_d` as much as possible without introducing jittering in the :term:`system response`. It should help reduce some of the if present.
+
+Adjust the setpoint up and down. Now, the arm should exhibit good behavior - quickly and precisely approaching the :term:`setpoint`.
+
+.. raw:: html
+
+   <details>
+     <summary>Tuning Solution</summary>
+
+
+In this particular example, reasonable values for the constants are :math:`K_{cosFF} = 5.92465`, :math:`K_p = 6.0`, and :math:`K_d = 2.0`. These should produce good results at all setpoints.
+   
+.. raw:: html
+
+   </details> <br>
+
+
+This shows how adding a carefully-chosen feed forward not only simplifies the calibration process, but produces better behavior at a wide range of setpoints.
+
+Common Issues
+-------------
+
+There are a number of common issues which can arise while tuning PID controllers.
+
+Integral Term Windup
+^^^^^^^^^^^^^^^^^^^^
 
 Beware that if :math:`K_i` is too large, integral windup can occur. Following a large change in :term:`setpoint`, the integral term can accumulate an error larger than the maximal :term:`control input`. As a result, the system overshoots and continues to increase until this accumulated error is unwound.
 
-.. note:: The :ref:`SysId toolsuite <docs/software/pathplanning/system-identification/index:System Identification>` can be used to model your system and give accurate Proportional and Derivative values. This is preferred over tuning the controller yourself.
+There are a few ways to mitigate this:
+
+1. Decrease the value of :math:`K_i`, down to zero if possible.
```

Comment:
Add: "Most mechanisms in FRC do not require any integral control, and systems that seem to require integral control to respond well probably have an inaccurate feedforward model."

Reply:
Yep, the above suggestion should probably be an ``.. important::`` admonition.